src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/cleanup/CleanUpJobService.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.hmcts.reform.wataskmonitor.services.jobs.cleanup;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmonitor.clients.CamundaClient;
+import uk.gov.hmcts.reform.wataskmonitor.config.job.CleanUpJobConfig;
+import uk.gov.hmcts.reform.wataskmonitor.domain.camunda.HistoricCamundaTask;
+import uk.gov.hmcts.reform.wataskmonitor.domain.jobs.GenericJobOutcome;
+import uk.gov.hmcts.reform.wataskmonitor.domain.jobs.GenericJobReport;
+import uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum;
+import uk.gov.hmcts.reform.wataskmonitor.utils.ResourceUtility;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.JobName.TASK_CLEAN_UP;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.ACTIVE_PROCESS_DELETE_REQUEST;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.CAMUNDA_CLEAN_UP_TASK_QUERY;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.HISTORIC_PROCESS_DELETE_REQUEST;
+
+@Component
+@Slf4j
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class CleanUpJobService {
+
+    public static final String CAMUNDA_DATE_TIME_REQUEST_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    public final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(CAMUNDA_DATE_TIME_REQUEST_PATTERN);
+
+    private final CamundaClient camundaClient;
+    private final CleanUpJobConfig cleanUpJobConfig;
+
+    @Autowired
+    public CleanUpJobService(CamundaClient camundaClient,
+                             CleanUpJobConfig cleanUpJobConfig) {
+        this.camundaClient = camundaClient;
+        this.cleanUpJobConfig = cleanUpJobConfig;
+    }
+
+    public boolean isAllowedEnvironment() {
+        log.info("{} cleanUpJobConfig: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig);
+
+        if (!cleanUpJobConfig.getAllowedEnvironment()
+            .contains(cleanUpJobConfig.getEnvironment().toLowerCase(Locale.ROOT))) {
```

Comment:
I think this requires a specific hard-coded block on prod being an allowed environment.  Even if someone goes and makes a mistake to configure it this code should not allow it to execute against prod.

The impact is potentially so significant we can't really leave it to chance.

Reply:
There may be other places where we'd want this type of check too.

Reply:
added this block after if statement for both environment check methods. is it enough?

if("prod".equalsIgnoreCase(cleanUpJobConfig.getEnvironment())){
            log.info("{} job can not be working on PROD environment", TASK_CLEAN_UP.name());
            return false;
        }

Reply:
This is ok.
You could have wrapped it into your isAllowedEnvironment method.